agent/internal/task_config.go

```diff
@@ -80,7 +80,8 @@ func (t *TaskConfig) GetExecTimeout() int {
 }
 
 // NewTaskConfig validates that the required inputs are given and populates the
-// information necessary for a task to run.
+// information necessary for a task to run. It is generally preferred to use
```

Comment:
Out of curiosity, are there still places where we're initializing the struct manually?

Reply:
It's only in tests. The problem is that when they do this, the risk of the test panicking is pretty high if we change the task running logic (e.g. if we use a new field that is populated for tasks in prod but the unit test forgets to set it because it hard-codes `taskConfig`). Using `internal.NewTaskConfig` instead of hard-coding `taskConfig` makes it harder for the tests to panic due to a slight change in assumptions.